src/intl/IntlDateTimeFormat.cpp

```diff
@@ -1011,20 +965,12 @@ UTF16StringDataNonGCStd IntlDateTimeFormatObject::format(ExecutionState& state,
 
     // Delegate remaining steps to ICU.
     UErrorCode status = U_ZERO_ERROR;
-    UTF16StringDataNonGCStd result;
-    result.resize(defaultStringBufferSize);
-    auto resultLength = udat_format(udat, x, (UChar*)result.data(), result.size(), nullptr, &status);
-    result.resize(resultLength);
-
-    if (status == U_BUFFER_OVERFLOW_ERROR) {
-        status = U_ZERO_ERROR;
-        udat_format(udat, x, (UChar*)result.data(), resultLength, nullptr, &status);
-    }
-    if (U_FAILURE(status)) {
+    auto formatResult = INTL_ICU_STRING_BUFFER_OPERATION_COMPLEX(udat_format, nullptr, m_icuDateFormat, x);
```

Comment:
If you were expecting `status` to be used inside a macro, this is wrong. 
It is re-declared with the same name within the macro.

Reply:
You are right. status is redundant variable. I have to remove it

Reply:
you use 'status' at 969 line

Reply:
we need only one status variable from INTL_ICU_STRING_BUFFER_OPERATION_COMPLEX